src/server/app/util/services/util.server.service.js

```diff
@@ -116,10 +116,20 @@ module.exports.toLowerCase = function (v){
 };
 
 module.exports.dateParse = function (date) {
-	if (null == date)
+	if (_.isNil(date)) {
 		return null;
+	}
+	if(_.isDate(date)) {
+		return date.getTime();
+	}
+	if(_.isFinite(date)) {
+		return date;
+	}
+	if(_.isString(date)) {
+		return Date.parse(date);
+	}
 
-	return Date.parse(date);
```

Comment:
Do we want to leave this as the default case? I'm thinking that way if an object other than the types checked above is given, we can default to Date.parse, which I think will call toString on the object if it exists.